### PR TITLE
Add bugfix to MaP IoU computation

### PR DESCRIPTION
**Pull request by LukeWood**

### CODE REVIEW

File: keras_cv/metrics/coco/mean_average_precision.py [hunk 1]
```diff
@@ -229,7 +229,7 @@ def update_state(self, y_true, y_pred, sample_weight=None):
                 )
 
                 ious = iou_lib.compute_iou(
-                    ground_truths_by_category, detections_by_category, "yxyx"
+                    ground_truths_by_category, detections_by_category, "xyxy"
```
Comment (ianstenbit) [inline keras_cv/metrics/coco/mean_average_precision.py:232]: Is there a test case we could write which would have caught this?

**Review by ianstenbit — APPROVED** [review]

File: keras_cv/metrics/coco/mean_average_precision.py (hunk 1 above)
Comment (bhack) [reply keras_cv/metrics/coco/mean_average_precision.py:232]: As bboxes argument/objects don't transport any metadata about the format it is hard.

https://github.com/keras-team/keras-cv/blob/3a9b8f020ba499af590ccfffee6603bc1a4c0764/keras_cv/bounding_box/iou.py#L65-L97

You could check if x <= height  and y <= width but just in the case we don't have out of "frame" coordinates.

Comment (tanzhenyu) [reply keras_cv/metrics/coco/mean_average_precision.py:232]: +1

**Comment by LukeWood** [comment]
Removed in favor of the final OD COCO metric PyMetric approach

**Pull request closed without merging**